src/dmd/visitor.h

```diff
@@ -557,6 +557,12 @@ class Visitor
     virtual void visit(StructInitializer *i) { visit((Initializer *)i); }
     virtual void visit(ArrayInitializer *i) { visit((Initializer *)i); }
     virtual void visit(VoidInitializer *i) { visit((Initializer *)i); }
+};
+
+class Visitor : public ParseTimeVisitor
+{
+public:
+    using ParseTimeVisitor::visit;
```

Comment:
Is this really necessary?

Reply:
Yep, at least for C++11+ compilers, otherwise there are warnings as the base class overloads are hidden.

Reply:
Hmm, that's what I'm worried about. Breaking C++98

Reply:
Eh, just switch to using D already :+1:

Reply:
This visitor class is used by all codegen routines, it can't be in any other language. Breaking compatibility is not an option. I think it bad enough that this class was rewritten for the sake of trying to make the frontend a better library. Had I not spotted, it would be @kinke having to spend a day trying to work out why nothing works at runtime, then doing 300 lines worth of rewriting the Visitor classes.

Sometimes I think it's forgotten that there are two major projects written in C++ that have been using DMD as if it was a library for over 14 years now.

Reply:
Yeah thanks a lot Iain for that, I know how tedious figuring out header divergences are.

Wrt. the `using`, I don't know how to express it in pre-C++11 or whether it's even required.

Reply:
It should be fine, as the testsuite compiles with `-std=c++98`.  I'm just being overly cautious here, and I feel I at least have good reason to be.